app/src/main/res/navigation/navigation_graph.xml

```diff
@@ -3,34 +3,34 @@
     xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
     android:id="@+id/navigation_graph"
-    app:startDestination="@id/authenticationFragment">
+    app:startDestination="@id/login_navigation">
 
     <fragment
-        android:id="@+id/authenticationFragment"
-        android:name="com.picone.go4lunch.presentation.ui.AuthenticationFragment"
-        android:label="fragment_authentication"
-        tools:layout="@layout/fragment_authentication">
-        <action
-            android:id="@+id/action_authenticationFragment_to_mapsFragment"
-            app:destination="@id/mapsFragment" />
-    </fragment>
-
+        android:id="@+id/listFragment"
+        android:name="com.picone.go4lunch.presentation.ui.ListFragment"
+        android:label="fragment_list"
+        tools:layout="@layout/fragment_list" />
     <fragment
         android:id="@+id/mapsFragment"
         android:name="com.picone.go4lunch.presentation.ui.MapsFragment"
         android:label="fragment_maps"
         tools:layout="@layout/fragment_maps" />
-
-    <fragment
-        android:id="@+id/listFragment"
-        android:name="com.picone.go4lunch.presentation.ui.ListFragment"
-        android:label="fragment_list"
-        tools:layout="@layout/fragment_list" />
-
     <fragment
         android:id="@+id/restaurantDetailFragment"
         android:name="com.picone.go4lunch.presentation.ui.RestaurantDetailFragment"
         android:label="fragment_restaurant_detail"
         tools:layout="@layout/fragment_restaurant_detail" />
+    <navigation
+        android:id="@+id/login_navigation"
+        app:startDestination="@id/authenticationFragment">
+        <fragment
+            android:id="@+id/authenticationFragment"
+            android:name="com.picone.go4lunch.presentation.ui.AuthenticationFragment"
+            android:label="fragment_authentication"
+            tools:layout="@layout/fragment_authentication"/>
+        <action
```

Comment:
You don't need action to maps here. 
Use `popBackStack()` instead.
See : [https://developer.android.com/guide/navigation/navigation-conditional](url)

Reply:
I've seen that popBackStack was deprecated in API 28, and in the link you've given to me, they're using ViewModel. Is this a better way to do navigation? If yes i'll done this when i'll pass all my methods to view model.